snap/validate.go

```diff
@@ -468,8 +476,11 @@ func ValidateLayout(layout *Layout, blacklist []string) error {
 	if !isAbsAndClean(mountPoint) {
 		return fmt.Errorf("layout %q uses invalid mount point: must be absolute and clean", layout.Path)
 	}
+
+	// effective path contains a trailing slash for directory-like mount entries.
+	effectivePath := layout.effectivePath()
 	for i := range blacklist {
-		if strings.HasPrefix(mountPoint, blacklist[i]) {
+		if (strings.HasSuffix(blacklist[i], "/") && strings.HasPrefix(effectivePath, blacklist[i])) || effectivePath == blacklist[i] {
```

Comment:
Yeah, some sort of helper might make this easier to read

Reply:
I implemented a helper I discussed with @chipaca now.